lab-judy/test/gallery-service-test.js

```diff
@@ -0,0 +1,185 @@
+      });
+
+      this.$httpBackend.flush();
+    });
+  });
+
+  //GET /api/login (authService)
+  describe('testing GET user login at /api/login', () => {
+    it('should login an existing user', () => {
+      let userData = {
+        name: 'Judy',
+        username: 'fakeUser',
+        password: 'ABCD',
+      };
+
+      let base64 = this.$window.btoa(`${userData.username}:${userData.password}`);
+
+      let headers = {
+        Accept: 'application/json',
+        Authorization: `Basic ${base64}`,
+      };
+
+      this.$httpBackend.expectGET('http://localhost:3000/api/login', headers)
+      .respond(200, '1234');
+
+      this.authService.login(userData)
+      .then((token) => {
+        expect(token).toBe('1234');
+      });
+
+      this.$httpBackend.flush();
+    });
+  });
+
+
+  //GET /api/gallery
+  describe('testing GET a user\'s galleries at /api/gallery', () => {
+    it('should return a user\'s galleries as array', () => {
+
+      let galleries = [
+        {
+          name: 'exampleGallery1',
+          desc: 'yohoo',
+        },
+        {
+          name: 'exampleGallery2',
+          desc: 'wee',
+        },
+      ];
+
+      let headers = {
+        Authorization: 'Bearer 1234',
+        Accept: 'application/json',
+      };
+
+      this.$httpBackend.expectGET('http://localhost:3000/api/gallery/?sort=dsc', headers)
+      .respond(200, galleries);
+
+      this.galleryService.fetchGalleries()
+      .then((res) => {
+        expect(res).toEqual(galleries);
+      });
+
+      this.$httpBackend.flush();
+    });
+  });
+
+  //PUT /api/gallery/galleryID
+  describe('testing PUT to /api/gallery/galleryID', () => {
+    it('should update a user\'s gallery', () => {
+
+      let gallery = {
+        name: 'exampleGallery1',
+        desc: 'yohoo',
+      };
+
+
+      let headers = {
+        Authorization: 'Bearer 1234',
+        Accept: 'application/json',
+        'Content-Type':'application/json',
+      };
+
+      this.$httpBackend.expectPUT('http://localhost:3000/api/gallery/helloworld', gallery, headers)
+      .respond(200, {_id: 'helloworld', username: 'slugbyte',  name: gallery.name, desc: gallery.desc, pics: []});
+
+      this.galleryService.updateGallery('helloworld', gallery)
+      .then((res) => {
+        expect(res).toEqual({_id: 'helloworld', username: 'slugbyte',  name: gallery.name, desc: gallery.desc, pics: []});
+      });
+
+      this.$httpBackend.flush();
+    });
+  });
+  //DELETE /api/gallery/galleryID
+  describe('testing DELETE to /api/gallery/galleryID', () => {
+    it('should delete a user\'s gallery', () => {
+
+      let headers = {
+        Accept: 'application/json',
+        Authorization: 'Bearer 1234',
+      };
+
+      this.$httpBackend.expectDELETE('http://localhost:3000/api/gallery/helloworld', headers)
+      .respond(204);
+
+      this.galleryService.deleteGallery('helloworld')
+      .then((res) => {
+        expect(res).toEqual(undefined);
```

Comment:
This works, but it's kind of an odd test. You could resolve something with $q in the deleteGallery method and then test that you get the resolved value back here if you wanted to refactor.